src/dispensary.js

```diff
@@ -0,0 +1,218 @@
+      })
+      .catch((err) => {
+        console.error('ERROR', err);
+      });
+  }
+
+  getLibraries() {
+    if (this._libraries !== null) {
+      return Promise.resolve(this._libraries);
+    }
+
+    try {
+      var libraryJSON = fs.readFileSync(this.libraryFile);
+      this._libraries = JSON.parse(libraryJSON);
+      return Promise.resolve(this._libraries);
+    } catch (err) {
+      if (err.toString().match(/^SyntaxError/)) {
+        return Promise.reject(new Error(
+          `JSONError: ${this.libraryFile} is not valid JSON.`));
+      }
+
+      return Promise.reject(new Error(
+        `${this.libraryFile} does not exist or is not a file.`));
+    }
+  }
+
+  _getAllFilesFromLibrary(library) {
+    var files = [];
+
+    for (let version of library.versions) {
+      if (library.filename) {
+        files.push({
+          file: library.filename,
+          fileOut: library.filenameOutput || library.filename,
+          library: library,
+          version: version,
+        });
+      }
+
+      if (library.filenameMinified) {
+        files.push({
+          file: library.filenameMinified,
+          fileOut: library.filenameMinifiedOutput || library.filenameMinified,
+          library: library,
+          version: version,
+        });
+      }
+    }
+
+    return files;
+  }
+
+  getFiles(libraries) {
+    return new Promise((resolve) => {
+      var files = [];
+
+      var queue = async.queue(this._getFile, this.maxHTTPRequests);
+      queue.drain = function() {
+        log.debug('All downloads completed.');
+        resolve(libraries);
+      };
+
+      for (let i in libraries) {
+        let library = libraries[i];
+
+        if (!library.files) {
+          library.files = [];
+        }
+
+        files = files.concat(this._getAllFilesFromLibrary(library));
+      }
+
+      queue.push(files);
+    });
+  }
+
+  _getFile(fileInfo, callback) {
+    var url = fileFormat(fileInfo.library.urlMin || fileInfo.library.url, {
+      file: fileInfo.file,
+      version: fileInfo.version,
+    });
+
+    log.debug(`Requesting ${url}`);
+
+    request.get({url: url}, (err, response, data) => {
+      if (err || !response) {
+        log.error(`${url} encountered an error: ${err}.`);
+        return callback(new Error(err));
+      }
+
+      if (response && response.statusCode !== 200) {
+        log.warn(`${url} produced code ${response.statusCode}`);
+        return callback();
+      }
+
+      log.debug(`Downloaded ${url}`);
+
+      fileInfo.library.files.push({
+        contents: data,
+        file: fileInfo.file,
+        fileOut: fileInfo.fileOut,
+        version: fileInfo.version,
+      });
+
+      callback();
+    });
+  }
+
+  getHashes(libraries) {
+    for (let library of libraries) {
+      for (let i in library.files) {
+        library.files[i].hash = hasher(library.files[i].contents);
+      }
+    }
+
+    return Promise.resolve(libraries);
+  }
+
+  getVersions(libraries) {
```

Comment:
Can this be refactored out to just call `getVersion()` instead of `this.getVersion()`?